Microsoft.RestApi/Transformers/RestOperationTransformer.cs

```diff
@@ -257,5 +257,119 @@ private static IList<DefinitionEntity> TransformDefinitions(OpenApiDocument open
+            {
+                var keyValue = openApiSecurityRequirement.Single();
+                var openApiSecurityScheme = keyValue.Key;
+                var flows = new List<FlowEntity>();
+
+                OpenApiOAuthFlow openApiOAuthFlow;
+                if ((openApiOAuthFlow = openApiSecurityScheme.Flows.AuthorizationCode) != null)
+                {
+                    flows.Add(NewFlowEntity("authorizationCode", keyValue.Value, openApiOAuthFlow));
+                }
+
+                if ((openApiOAuthFlow = openApiSecurityScheme.Flows.ClientCredentials) != null)
+                {
+                    flows.Add(NewFlowEntity("clientCredentials", keyValue.Value, openApiOAuthFlow));
+                }
+
+                if ((openApiOAuthFlow = openApiSecurityScheme.Flows.Implicit) != null)
+                {
+                    flows.Add(NewFlowEntity("implicit", keyValue.Value, openApiOAuthFlow));
+                }
+
+                if ((openApiOAuthFlow = openApiSecurityScheme.Flows.Password) != null)
+                {
+                    flows.Add(NewFlowEntity("password", keyValue.Value, openApiOAuthFlow));
+                }
+
+
+                var securityEntity = new SecurityEntity
+                {
+                    Type = openApiSecurityScheme.Type.ToString(),
+                    Description = openApiSecurityScheme.Description,
+                    In = openApiSecurityScheme.In.ToString(),
+                    Flows = flows
+                };
+                securities.Add(securityEntity);
+            }
+
+            return securities;
+        }
+
+        public static FlowEntity NewFlowEntity(string name, IList<string> scopeNames, OpenApiOAuthFlow openApiOAuthFlow)
+        {
+            var scopes = new List<SecurityScopeEntity>();
+            foreach (var scopeName in scopeNames)
+            {
+                var scope = new SecurityScopeEntity
+                {
+                    Name = scopeName,
+                    Description = openApiOAuthFlow.Scopes.Where(s => s.Key.Equals(scopeName)).Single().Value
+                };
+                scopes.Add(scope);
+            }
+
+            return new FlowEntity
+            {
+                Name = name,
+                AuthorizationUrl = openApiOAuthFlow.AuthorizationUrl.ToString(),
+                TokenUrl = openApiOAuthFlow.TokenUrl.ToString(),
+                Scopes = scopes
+            };
+        }
+
+        public static IList<SeeAlsoEntity> TransformExternalDocs(OpenApiOperation openApiOperation)
+        {
+            var seeAlsoEntities = new List<SeeAlsoEntity>();
+            if (openApiOperation.ExternalDocs != null)
+            {
+                var seeAlsoEntity = new SeeAlsoEntity
+                {
+                    Description = openApiOperation.ExternalDocs.Description,
+                    Url = openApiOperation.ExternalDocs.Url.ToString()
```

Comment:
can this openApiOperation.ExternalDocs.Url be null? If yes, we could use openApiOperation.ExternalDocs.Url?.ToString()

Reply:
Url field is required in the openApi spec.

Reply:
ok